Update build docs (#202)
larq/compute-engine: docs/build_arm.md

# Building Larq Compute Engine for ARM-based systems
This page descibes how to build Larq Compute Engine (LCE) applications
for 32-bit, as well as 64-bit ARM-based systems. You can either build LCE
natively on your ARM device or cross-compile it from a non-ARM machine.
[Bazel](https://bazel.build/) is the primary build system for LCE, and it can
be used to cross-compile a binary for ARM-based systems using a non-ARM host.
To natively build on an ARM system itself we provide a script that uses the

> Makefile build system.

Reviewer: I am strongly against using the word "application". Application has different meaning (see [here](https://en.wikipedia.org/wiki/Application_software)). The right word here is "binary" or "executable". Even the TF lite benchmark guide use the word "binary".

> Makefile build system.

Author: I'll change it back to 'binary'.

I did have that particular meaning of the word application in mind, because the developers that write a program with LCE might want to use it to build an actual Android app, for example where LCE processes the camera input. In that case, application would be correct I think.

This leaves us with three ways to build LCE applications, which we recommend in
the following order:

> 1. To cross-compile LCE from a host machine, see the section

Reviewer: ```suggestion
1. To cross-compile LCE from a host machine, see section
```

Author: I think in this case we have to leave out the word 'section'. When there's a number, then you leave out the "the", for example "see Section 3".

   [Cross-compiling LCE with Bazel](#cross-compiling-lce-with-bazel)

> 2. To natively compile LCE, see the section

Reviewer: ```suggestion
2. To natively compile LCE, see section
```

>    install Bazel, see the section

Reviewer: ```suggestion
   install Bazel, see section
```

   [Cross-compiling LCE with Make](#cross-compiling-lce-with-make).

This guide will show you how to build the [LCE example application](../examples/lce_minimal.cc)

> and the [LCE benchmark tool](../larq_compute_engine/tflite/benchmark).

Reviewer: I don't think we need to give two examples here. just showing  example LCE application is enough. Building LCE benchmark tool does not add any new values or insight.

See [here](./inference.md) to find out how you can create your own LCE
inference application.

> Although the Raspberry Pi 3 and Raspberry Pi 4 have 64-bit CPUs, note that the

Reviewer: ```suggestion
NOTE: the Raspberry Pi 3 and Raspberry Pi 4 have 64-bit CPUs, note that the
```

> popular distribution Raspbian for the Raspberry Pi is a 32-bit OS. In order to

Reviewer: ```suggestion 
officially supported Raspberry Pi operating system "Raspbian" is a 32-bit OS. In order to
```

First configure Bazel using the instructions [here](build.md). Make sure you

> have run the `./configure.sh` script as instructed.

Reviewer: ```suggestion
First configure Bazel using the instructions [here](build.md#configure-bazelrc).
```

First configure Bazel using the instructions [here](build.md). Make sure you
have run the `./configure.sh` script as instructed.

To cross-compile the LCE example application for ARM architectures, the bazel
target needs to be built with the `--config=rpi3` (32-bit ARM) or

> `--config=aarch64` (64-bit ARM) flag. For example, to build the example

Reviewer: why do we need to have two different examples here? The example itself is not what matters, its about building and the building for both examples, beside just one flag, is the same.

Author: True. I'll just leave the `lce_minimal` example and then in the text state the name of the benchmark bazel target.
